lib/arel-helpers/join_association.rb

```diff
@@ -248,6 +250,50 @@ def join_association_6_0_0(table, association, join_type, options = {})
         end
       end
 
+      def join_association_6_1_0(table, association, join_type, options = {})
+        aliases = options.fetch(:aliases, []).index_by(&:table_name)
+        associations = association.is_a?(Array) ? association : [association]
+
+        alias_tracker = ActiveRecord::Associations::AliasTracker.create(
+          table.connection, table.name, {}
+        )
+
+        join_dependency = ActiveRecord::Associations::JoinDependency.new(
+          table, table.arel_table, associations, join_type
+        )
+
+        constraints = join_dependency.join_constraints([], alias_tracker, [])
+
+        constraints.map do |join|
+          apply_aliases(join, aliases)
```

Comment:
This is the major change between 6.0 and 6.1. In 6.0, the alias was applied by directly setting the table alias via `join.left.table_alias = found_alias.name`.

However in 6.1, the alias persists between specs, likely due to internal activerecord memoization which causes other specs that do not alias to fail.

I'm not entirely confident of the implementation here but the specs are all passing.

Reply:
Interesting, AR strikes again. Is there an internal AR method we can call instead of this `apply_aliases` method? I'm concerned `apply_aliases` isn't covering 100% of the possible node types.

Reply:
Ok, so after doing some poking around, I think the `apply_aliases` approach is likely the best way to go. I haven't been able to figure out how else to add in table aliases.